pkg/fake/utils.go

```diff
@@ -90,15 +90,34 @@ func SubnetsFromFleetRequest(createFleetInput *ec2.CreateFleetInput) []string {
 // Filters are chained with a logical "AND"
 func FilterDescribeSecurtyGroups(sgs []ec2types.SecurityGroup, filters []ec2types.Filter) []ec2types.SecurityGroup {
 	return lo.Filter(sgs, func(group ec2types.SecurityGroup, _ int) bool {
-		return Filter(filters, *group.GroupId, *group.GroupName, group.Tags)
+		return Filter(filters, *group.GroupId, *group.GroupName, "", group.Tags)
 	})
 }
 
 // FilterDescribeSubnets filters the passed in subnets based on the filters passed in.
 // Filters are chained with a logical "AND"
 func FilterDescribeSubnets(subnets []ec2types.Subnet, filters []ec2types.Filter) []ec2types.Subnet {
 	return lo.Filter(subnets, func(subnet ec2types.Subnet, _ int) bool {
-		return Filter(filters, *subnet.SubnetId, "", subnet.Tags)
+		return Filter(filters, *subnet.SubnetId, "", "", subnet.Tags)
+	})
+}
+
+func FilterDescribeCapacityReservations(crs []ec2types.CapacityReservation, ids []string, filters []ec2types.Filter) []ec2types.CapacityReservation {
+	idSet := sets.New[string](ids...)
+	return lo.Filter(crs, func(cr ec2types.CapacityReservation, _ int) bool {
+		if len(ids) != 0 && !idSet.Has(*cr.CapacityReservationId) {
+			return false
+		}
+		if stateFilter, ok := lo.Find(filters, func(f ec2types.Filter) bool {
```

Comment:
I'm assuming it's too complicated to collapse all of the state filtering across all resources into the `Filter` function?

Reply:
It probably isn't, now that it's being used on a few types I can take a look